src/main/java/io/appform/dropwizard/actors/base/Handler.java

```diff
@@ -85,7 +92,30 @@ public void handleDelivery(String consumerTag, Envelope envelope,
         }
     }
 
-    private MessageMetadata messageProperties(final Envelope envelope) {
-        return new MessageMetadata(envelope.isRedeliver());
+    private Callable<Boolean> getHandleCallable(Envelope envelope,
+                                                AMQP.BasicProperties properties,
+                                                byte[] body) throws IOException {
+        val delayInMs = getDelayInMs(properties);
+        val expired = isExpired(properties);
+        val message = mapper.readValue(body, clazz);
+        return () -> handle(message, messageProperties(envelope, delayInMs), expired);
+    }
+
+    private Long getDelayInMs(AMQP.BasicProperties properties) {
```

Comment:
In case there is clock skew of few seconds between producer and consumer (with consumer clock behind producer), we would end up with negative `delayInMs`. What is the expected behaviour in such scenarios ?

Reply:
True but won't all our servers have a common time synced from a central server?

If not then even the expiryTTL can result in the message with an early expiry than the actual if the consumer clock is ahead by a few seconds

Reply:
We can assume a realistic clock skew (few seconds) but it's not possible to eliminate clock skew completely

Reply:
As discussed, setting delay to 0 in case of clock skew